File: app/src/main/java/com/arman/queuetube/util/services/KillNotificationService.java
package com.arman.queuetube.util.services;

import android.app.NotificationManager;
import android.app.Service;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;

import com.arman.queuetube.config.Constants;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.app.JobIntentService;

public class KillNotificationService extends JobIntentService {

    private KillBinder binder;

    public KillNotificationService() {
        this.binder = new KillBinder(this);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return this.binder;
    }

    @Override
    protected void onHandleWork(@NonNull Intent intent) {

    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        return Service.START_STICKY;
    }

    private void clearNotification() {
        NotificationManager notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
        notificationManager.cancel(Constants.Notification.ID);
    }

    @Override
    public void onTaskRemoved(Intent rootIntent) {
        super.onTaskRemoved(rootIntent);
        clearNotification();
    }

    public class KillBinder extends Binder {

        public Service service;

        public KillBinder(Service service) {
            this.service = service;
        }

    }

}

File: app/src/main/java/com/arman/queuetube/util/services/KillNotificationService.kt
package com.arman.queuetube.util.services

import android.app.NotificationManager
import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import androidx.core.app.JobIntentService
import com.arman.queuetube.config.Constants

class KillNotificationService : JobIntentService() {

    private val binder: KillBinder

    init {
        this.binder = KillBinder(this)
    }

    override fun onBind(intent: Intent): IBinder? {
        return this.binder
    }

    override fun onHandleWork(intent: Intent) {

    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        return Service.START_STICKY
    }

    private fun clearNotification() {
        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        notificationManager.cancel(Constants.Notification.ID)
    }

    override fun onTaskRemoved(rootIntent: Intent) {
        super.onTaskRemoved(rootIntent)
        clearNotification()
    }

    inner class KillBinder(var service: Service) : Binder()

}
